source/change-streams/change-streams.rst

```diff
@@ -417,13 +464,17 @@ Once a ``ChangeStream`` has encountered a resumable error, it MUST attempt to re
 - If the ``ChangeStream`` has not received any changes, and ``resumeAfter`` is not specified, and the max wire version is >= ``7``:
 
     - The driver MUST execute the known aggregation command.
-    - The driver MUST specify the ``startAtOperationTime`` key set to the ``startAtOperationTime`` provided by the user or saved from the original aggregation.
+    - If the ``ChangeStream`` has a saved ``minPromisedResumeToken``:
+
+         - The driver MUST specify ``resumeAfter`` with the cached ``minPromisedResumeToken``.
+    - Else:
+
+         - The driver MUST specify the ``startAtOperationTime`` key set to the ``startAtOperationTime`` provided by the user or saved from the original aggregation.
```

Comment:
If we didn't have to support pre 4.2 versions we could simplify the resume logic to omit ever using startAtOperationTime.

Reply:
I don't have a strong opinion on whether we check wire version. But if we don't check wire version, we could foreseeably backport the server change to 4.0 and have new drivers use it correctly.

Reply:
I don't see the need to check wire version, as I expect any driver implementations could simply check the response document for `aggregate` or `getMore` for a `promisedMinResumeToken` key (if one exists).

Reply:
@dstorch: I agree that `startAtOperationTime` should become obsolete with `promisedMinResumeToken`, but please make sure drivers are informed when/if the server drops support for `startAtOperationTime`!